library/src/main/java/com/novoda/downloadmanager/LiteDownloadManagerDownloader.java

```diff
@@ -95,15 +94,18 @@ private void waitForDownloadService() {
     }
 
     private void executeDownload(final DownloadBatch downloadBatch) {
-        updateStatusToQueuedIfNeeded(downloadBatch);
+        InternalDownloadBatchStatus downloadBatchStatus = downloadBatch.status();
+        if (downloadBatchStatus.status() == DOWNLOADED) {
+            return;
+        }
+
+        updateStatusToQueuedIfNeeded(downloadBatchStatus);
         downloadService.download(downloadBatch, downloadBatchCallback());
     }
 
-    private void updateStatusToQueuedIfNeeded(DownloadBatch downloadBatch) {
-        InternalDownloadBatchStatus liteDownloadBatchStatus = downloadBatch.status();
-
-        if (liteDownloadBatchStatus.status() != PAUSED) {
-            liteDownloadBatchStatus.markAsQueued(downloadsBatchPersistence);
+    private void updateStatusToQueuedIfNeeded(InternalDownloadBatchStatus downloadBatchStatus) {
+        if (downloadBatchStatus.status() != PAUSED) {
```

Comment:
if the status is `ERROR`  should it be Queued?

Reply:
Yes, it should be re-queued as it might have been a network error.